### ISSUE TITLE
manually set vaccination rate

### ISSUE THREAD

**Issue by rmbellovin** [microCOVID/microCOVID#889]
If you toggle "Override location-based data", it should be possible to manually set the vaccination rate for the location.

**Comment by twpol** [comment]
While the input box has appeared on https://www.microcovid.org/, I'm not convinced this feature is working. The "Percent of population fully vaccinated" value does not appear in the URL, like the other manual inputs, and the average person risk profile does not seem to change when I change the value:

![image](https://user-images.githubusercontent.com/1017843/132846769-d496b51c-ef0d-4ac9-b611-ed5f0bc92b63.png)

Have I misunderstood this feature request?

I want to be able to provide the vaccination percentage along with the other manual override values and have it taken into account just like it does when selecting a region with built-in vaccination data

**Comment by diomidov** [comment]
I agree with twpol and I think this issue should be reopened.

**Comment by diomidov** [comment]
And I think you should be able to enter it even if there is vaccination data available. For example my school has a much higher vaccination rate than the county it's in (98% vs 68%). If I know that people at my event are all students, it makes more sense to use the school data rather than the county data.

**Comment by rmbellovin** [comment]
> And I think you should be able to enter it even if there is vaccination data available. For example my school has a much higher vaccination rate than the county it's in (98% vs 68%). If I know that people at my event are all students, it makes more sense to use the school data rather than the county data.

While arguably you should be able to enter it even if there is vaccination data available, this is the wrong usage (and highlights why you should be careful about letting people override existing data).  The problem is that if you leave the covid prevalence rate set to the county's prevalence rate but change the vaccination rate to 98%, the calculator will show all activities as much higher risk than they actually are.  You should either override both the prevalence rate and the vaccination rate, or say that the people near you are fully vaccinated.

**Comment by GPPassos** [comment]
> The problem is that if you leave the covid prevalence rate set to the county's prevalence rate but change the vaccination rate to 98%, the calculator will show all activities as much higher risk, even though they're actually much _lower_ risk. 

Sorry, how so?

> You should either override both the prevalence rate and the vaccination rate, or say that the people near you are fully vaccinated.

The current interface is that if you check the box for overriding location-based data, it uses all parameters that you insert, including both prevalence rate and vaccination rate, so as I understand this is already the current issue.

**Comment by rmbellovin** [comment]
> > The problem is that if you leave the covid prevalence rate set to the county's prevalence rate but change the vaccination rate to 98%, the calculator will show all activities as much higher risk, even though they're actually much _lower_ risk.
> 
> Sorry, how so?

Let's say that in a given region the covid prevalence is P_covid, and let's say that P_vac is the proportion of the population that's vaccinated.  If I want to calculate the risk of doing a particular activity when only vaccinated people are around, I need to first calculate the covid prevalence P_covid,vac in the region among vaccinated people.  I'll have an equation 
P_covid = P_vac * P_covid,vac + (1 - P_vac) * P_covid,unvac
relating the total prevalence, P_covid,vac, and the prevalence P_covid,unvac among unvaccinated people.

For simplicity, suppose that half the people are vaccinated, so that P_covid is just the average of P_covid,vac and P_covid,unvac.  That's not quite enough information, but the vaccine efficacy will tell us how P_covid,vac and P_covid,unvac relate to each other.  If the vaccine is very effective, we'll find that P_covid,vac is much lower than P_covid and conclude that, say, dining indoors is much safer if everyone in the room is fully vaccinated.

Now suppose I say "everyone in my town is vaccinated, I'll set P_vac to 100%", but I use P_covid for the county.  Then I'll think that P_covid,vac = P_covid, which is wrong.

> 
> > You should either override both the prevalence rate and the vaccination rate, or say that the people near you are fully vaccinated.
> 
> The current interface is that if you check the box for overriding location-based data, it uses all parameters that you insert, including both prevalence rate and vaccination rate, so as I understand this is already the current issue.

Right now, you can override the prevalence numbers but you can only choose a vaccination rate value if none is known.  I'm pointing out that because there are subtle dependencies between these values, permitting people to freely override all of them makes it very easy for them to mislead themselves about exactly what they're calculating.

**Comment by diomidov** [comment]
> The problem is that if you leave the covid prevalence rate set to the county's prevalence rate but change the vaccination rate to 98%, the calculator will show all activities as much higher risk, even though they're actually much lower risk. You should either override both the prevalence rate and the vaccination rate, or say that the people near you are fully vaccinated.

Yes, I understand that. I'm trying to set *both* the covid prevalence and vaccination rate. I can override the former but not the latter:
![image](https://user-images.githubusercontent.com/22151995/134752482-07e39a55-b4e0-4cf5-8ed0-d9552c5a659f.png)

**Comment by blanchardjeremy** [comment]
Ah yes. This is still an issue. Tagging @oxytocinlove. 

I can reproduce (as stated above) by doing:
1. Select a state/city
2. Check the "override" box
3. The vaccination rate is readonly. 

This line: https://github.com/microCOVID/microCOVID/pull/969/files#diff-f3880ab005409eff454c09c0b6afef593647cfb40f820a6b3948d8f2528d354fR164

Should be set to check whether "override" is selected or not (rather than checking whether the value in the data model is null).

**Comment by BeeFox-sys** [comment]
boosting this issue as it is an important feature for countries that do not currently have stats on the site
